File: react_native/Hybrid/Hybrid/ViewController.m
//
//  ViewController.m
//  Hybrid
//
//  Created by Hirohisa Kawasaki on 4/25/15.
//  Copyright (c) 2015 Hirohisa Kawasaki. All rights reserved.
//

#import "ViewController.h"
#import "ReactViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cell"];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {

    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
    cell.textLabel.text = @"React ViewController";

    return cell;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 1;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {

    UIViewController *viewController = [[ReactViewController alloc] initWithNibName:@"ReactViewController" bundle:nil];
    [self.navigationController pushViewController:viewController animated:YES];
}

@end

File: swift4/RefreshControlSample/RefreshControlSample/ViewController.swift
//
//  ViewController.swift
//  RefreshControlSample
//
//  Created by Hirohisa Kawasaki on 2018/07/02.
//  Copyright © 2018 Hirohisa Kawasaki. All rights reserved.
//

import UIKit

class ViewController: PullToRefreshViewController, UITableViewDelegate, UITableViewDataSource {

    @IBOutlet weak var tableView: RefreshableTableView!

    var isAlphabet = false
    let alphabet = "abcdefghijklmnopqrstuvwxyz".map({$0})

    override func viewDidLoad() {
        super.viewDidLoad()

        tableView.refreshHeaderView.backgroundColor = UIColor.blue

        tableView.onRefresh = {
            print("refreshing")

            self.isAlphabet = !self.isAlphabet
            DispatchQueue.main.asyncAfter(deadline: .now() + 1, execute: {
                self.tableView.stopAnimating()
                self.tableView.reloadData()
            })
        }

        tableView.onLoadMore = {
            print("loading more")
            self.tableView.stopAnimating()
        }

        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
    }

    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
        let number = indexPath.row
        if isAlphabet {
            cell.textLabel?.text = String(alphabet[number])
        } else {
            cell.textLabel?.text = number.description
        }
        return cell
    }

    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 20
    }
}

